compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -13787,7 +13787,7 @@ OMR::Z::TreeEvaluator::iRegStoreEvaluator(TR::Node * node, TR::CodeGenerator * c
 
    if (!useLGHI)
       {
-      globalReg = cg->evaluate(value);
+      globalReg = cg->gprClobberEvaluate(value);
```

Comment:
I think globalReg should only be clobberable when we are going to sign extend that explicitly through an instruction. 

In [this block](https://github.com/eclipse/omr/blob/09c77a981fea4b83427d7f18b26f9fa4c4faf50a/compiler/z/codegen/OMRTreeEvaluator.cpp#L13781-L13786) a properties in the node is set, so when we evaluate the node, it will sign extend, so in that case, clobber evaluate would be useless (Even we would allocate a register unnecessary). Also in that case, we would not even clobber it in this evaluator. 

We should have clobberable register in case, we are going to generate `LGFR` and it is not constant.

Reply:
Makes sense. So we can add a conditional check in this block to only clobber evaluate when an LGFR will be generated. Would the following work?

```
if (!useLGHI)
      {
      if (needsLGFR && noLGFgenerated)
         {
         globalReg = cg->clobberEvaluate(value);
         }
      else
         {
         globalReg = cg->evaluate(value);
         }
      }
```

This avoids clobber evaluating when the value node is an integer constant, and when sign extension will be generated by evaluating the child normally. Let me know and I'll update the PR with the change.

Reply:
Yes this would work, you can use conditional assignment to simplify the above code. 
```
globalReg = needsLGFR && noLGFgenerated ? cg->clobberEvaluate(value) : cg->evaluate(value);
```

Reply:
a97acb524be5b30a80b91a61e29495355f20d68d